controllers/subscriptions.go

```diff
@@ -285,6 +326,24 @@ func EnsureVendorCsv(cli client.Client, csvName string) (*operatorv1alpha1.Clust
 	}
 	_, err = controllerutil.CreateOrUpdate(context.TODO(), cli, csvObj, func() error {
 		csvObj.OwnerReferences = []metav1.OwnerReference{}
+
+		// Shut down the OCS client operator CSV pods in non provider mode
+		if csvName == OcsClientSubscriptionStartingCSV {
+			isProvider, err := isProviderMode(cli)
+			if err != nil {
+				return err
+			}
+
+			var replicas int32 = 0
+			if isProvider {
+				replicas = 1
+			}
+
+			for i := range csvObj.Spec.InstallStrategy.StrategySpec.DeploymentSpecs {
```

Comment:
Are we sure nothing in this long path is a pointer and has the possibility to be `nil`?
It any of it is a pointer, then consider having a nil check before accessing to make sure we do not crush the pod in runtime for some edge case.

Reply:
There is no pointer.

Reply:
Ok